### PR TITLE
feat: RBAC chain root and registrations queries

### PR DESCRIPTION
**Pull request by FelipeRosa**

### CODE REVIEW

File: catalyst-gateway/bin/src/service/api/cardano/rbac/registrations_get.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    },
+    service::common::responses::WithErrorResponses,
+};
+
+/// Endpoint responses.
+#[derive(ApiResponse)]
+pub(crate) enum Responses {
+    /// Success returns a list of registration transaction ids.
+    #[oai(status = 200)]
+    Ok(Json<Vec<String>>),
```
Comment (stevenj) [inline catalyst-gateway/bin/src/service/api/cardano/rbac/registrations_get.rs:21]: Should be a properly documented response structure, not just a string.

File: catalyst-gateway/bin/src/service/api/cardano/rbac/chain_root_get.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        stake_address: stake_address.to_vec(),
+    })
+    .await;
+
+    match query_res {
+        Ok(mut row_iter) => {
+            if let Some(row_res) = row_iter.next().await {
+                let row = match row_res {
+                    Ok(row) => row,
+                    Err(err) => return AllResponses::handle_error(&anyhow!(err)),
```
Comment (stevenj) [inline catalyst-gateway/bin/src/service/api/cardano/rbac/chain_root_get.rs:45]: Should be logged, and a 500 returned with no body.

File: catalyst-gateway/bin/src/service/api/cardano/rbac/chain_root_get.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                };
+
+                let chain_root_hex = hex::encode(row.chain_root);
+                Responses::Ok(Json(chain_root_hex)).into()
```
Comment (stevenj) [inline catalyst-gateway/bin/src/service/api/cardano/rbac/chain_root_get.rs:49]: Should return a response structure, not a string.

File: catalyst-gateway/bin/src/service/api/cardano/rbac/chain_root_get.rs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    #[oai(status = 404)]
+    NotFound,
+}
+
+pub(crate) type AllResponses = WithErrorResponses<Responses>;
+
+/// Get chain root endpoint.
+pub(crate) async fn endpoint(stake_address: StakeAddress) -> AllResponses {
+    let Some(session) = CassandraSession::get(true) else {
+        return AllResponses::handle_error(&anyhow!("Failed to connect to database"));
```
Comment (stevenj) [inline catalyst-gateway/bin/src/service/api/cardano/rbac/chain_root_get.rs:32]: Should be logged, and a 500 returned with no body.

File: catalyst-gateway/bin/src/service/api/cardano/rbac/chain_root_get.rs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        objects::cardano::stake_address::StakeAddress, responses::WithErrorResponses,
+    },
+};
+
+/// Endpoint responses.
+#[derive(ApiResponse)]
+pub(crate) enum Responses {
+    /// Success returns the chain root hash.
+    #[oai(status = 200)]
+    Ok(Json<String>),
```
Comment (stevenj) [inline catalyst-gateway/bin/src/service/api/cardano/rbac/chain_root_get.rs:21]: Should be a properly structured response, not just a string.

**Review by minikin — APPROVED** [review]
lgtm

**Pull request merged**